execution_engine/src/shared/opcode_costs.rs

```diff
@@ -0,0 +1,288 @@
+use std::collections::BTreeMap;
+
+use datasize::DataSize;
+use pwasm_utils::rules::{InstructionType, Metering, Set};
+use rand::{distributions::Standard, prelude::*, Rng};
+use serde::{Deserialize, Serialize};
+
+use casper_types::bytesrepr::{self, FromBytes, ToBytes, U32_SERIALIZED_LENGTH};
+
+const NUM_FIELDS: usize = 17;
+pub const OPCODE_COSTS_SERIALIZED_LENGTH: usize = NUM_FIELDS * U32_SERIALIZED_LENGTH;
+
+// Taken (partially) from parity-ethereum
+#[derive(Copy, Clone, PartialEq, Eq, Serialize, Deserialize, Debug, DataSize)]
+pub struct OpCodeCosts {
```

Comment:
```suggestion
pub struct OpcodeCosts {
```

No need to capitalize the `c` in `Opcode`:
https://en.wikipedia.org/wiki/Opcode